lib_vehicle_model/include/lib_vehicle_model/LibVehicleModel.h

```diff
@@ -0,0 +1,105 @@
+#pragma once
+/*
+ * Copyright (C) 2018-2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <memory>
+#include <stdexcept>
+#include <ros/ros.h>
+#include "ModelAccessException.h"
+#include "VehicleState.h"
+#include "VehicleMotionModel.h"
+#include "VehicleModelControlInput.h"
+#include "ParameterServer.h"
+#include "KinematicsSolver.h"
+#include "KinematicsProperty.h"
+
+namespace lib_vehicle_model {
+
+  //
+  // Public Namespace
+  //
+
+  /**
+   * @brief Initialization function for vehicle model. Loads the vehicle model as specified by parameters 
+   * 
+   * When provided with a NodeHandle this function uses the ROSParameterServer instead of a user specified parameter server
+   * 
+   * @param nh A pointer to the node handle which will be used to initialize the ROSParameterServer
+   * 
+   * @throws std::invalid_argument If the model could not be loaded or parameters could not be read
+   * @throws ModelAccessException If this function is called more than once within the same process execution
+   * 
+   */ 
+  void init(std::shared_ptr<ros::NodeHandle> nh);
+
+  /**
+   * @brief Initialization function for class. Loads the library as specified by parameters 
+   * 
+   * @param parameter_server A reference to the parameter server which vehicle models will use to load parameters
+   * 
+   * @throws std::invalid_argument If the model could not be loaded or parameters could not be read
+   * @throws ModelAccessException If this function is called more than once within the same process execution
+   * 
+   */ 
+  void init(std::shared_ptr<ParameterServer> parameter_server);
+
+  /**
+   * @brief Function to unload the currently loaded model
+   * 
+    * NOTE: Note this function is not required to be called for proper shutdown upon program completion.
+    * This should be used for special cases where the user is positive no other components still need the current model such as in unit testing
+    * 
+    */ 
+    void unload();
```

Comment:
Unit testing primarily, because the model is loaded as a global object it will persist across unit tests.

At one point I actually had this hidden using a named internal namespace, but I figured that added extra files while removing this as an option for the user if they wish to use it in their own unit tests.

Reply:
I think init and unload are driving the internal state of this namespace. For outside consumer, it might be difficult to know what the current state is. The result is that in everyplace, if we want to use make a function call to this library, we need to handle exceptions very carefully. I think it is a general issue which is not specific to this function.